signal/signal.go

```diff
@@ -175,12 +182,27 @@ func (c *Interceptor) mainInterruptHandler() {
 		select {
 		case signal := <-c.interruptChannel:
 			log.Infof("Received %v", signal)
+
+			// Set the interceptor exit code status to 0 once we get
+			// a message from interrupt Channel since it would
+			// exit successfully after graceful shut down when we
+			// close the quit channel.
+			c.setExitCode(0)
+
 			shutdown()
 
 		case <-c.shutdownRequestChannel:
 			log.Infof("Received shutdown request.")
+
+			// Set interceptor exit code status to 1 once we get a
+			// message from shutdownRequest Channel since it would
+			// exit with an error status after graceful shutdown
+			// when we close the quit channel.
+			c.setExitCode(1)
```

Comment:
hmmm im not sure this is correct. 

`shutdownRequestChannel` could be closed if the `StopDaemon` rpc method is called. If LND shuts down smoothly after that, then the exit code should be 0, right?

Reply:
perhaps instead `RequestShutdown` should have an error code param... so that call-sights can determine what the error code should be.... but then also, even if a call-sight indicates a `0` error code, the shutdown process itself should be able to downgrade that to a 1 ofc

Reply:
> shutdownRequestChannel could be closed if the StopDaemon rpc method is called. If LND shuts down smoothly after that, then the exit code should be 0, right?

Ah - you can see this in action if you take a look at the failing itests. They all call `StopDaemon` to shutdown the various nodes but now exit code of each process is 1